packages/api/src/context.ts

```diff
@@ -473,62 +508,107 @@ export function tokenFromRequest(req: IncomingMessage | null): string | null {
   return null
 }
 
+/**
+ * Function that generate the key for the cache
+ * @param params
+ */
+
+function generateCacheKey(params: peerQueryParams) {
+  return (
+    crypto
+      // Hash function doesn't have to be crypto safe, just fast!
+      .createHash('md5')
+      .update(
+        `${JSON.stringify(params.hostURL)}${JSON.stringify(
+          params.variables?._v0_id
+        )}${JSON.stringify(params.query)}`
+      )
+      .digest('hex')
+  )
+}
+
+/**
+ * Function to that refreshes and initializes entries in the cache
+ * @param params
+ */
+
+async function loadFreshData(params: peerQueryParams) {
+  try {
+    const abortController = new AbortController()
+
+    // Since we use auto refresh cache we can safely set the timeout to 3sec
+    setTimeout(() => abortController.abort(), 3000)
+
+    const fetchResult = await fetch(params.hostURL, {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json', Authorization: `Bearer ${params.token}`},
+      body: JSON.stringify({
+        query: params.query,
+        variables: params.variables,
+        operationName: params.operationName
+      }),
+      signal: abortController.signal
+    })
+    const res = await fetchResult.json()
+    if (fetchResult?.status != 200) {
+      return {
+        errors: [new GraphQLError(`Peer responded with invalid status: ${fetchResult?.status}`)]
+      }
+    }
+    params.lastQueried = params.lastQueried ? params.lastQueried : new Date().getTime()
+    const cacheValue: peerCacheValue = {
+      data: res,
+      queryParams: params
+    }
+    peerCache.set(params.cacheKey, cacheValue)
+    return res
+  } catch (err) {
+    if (err.type === 'aborted') {
+      err = new Error(`Connection to peer (${params.hostURL}) timed out.`)
+    }
+
+    console.error(err)
```

Comment:
would use the logger() instead console log to be consistent.

Reply:
Basicaly just copy paste from prior code but i changed it